Compiler warnings

Leaving these here so they don't get overlooked:

```
[109/219] Building CXX object CMakeFiles/openloco.dir/src/OpenLoco/Paint/Paint.cpp.o
/home/aaron/Repositories/OpenLoco/OpenLoco/src/OpenLoco/Paint/Paint.cpp: In member function ‘OpenLoco::Ui::ViewportInteraction::InteractionArg OpenLoco::Paint::PaintSession::getNormalInteractionInfo(uint32_t)’:
/home/aaron/Repositories/OpenLoco/OpenLoco/src/OpenLoco/Paint/Paint.cpp:855:24: warning: potential null pointer dereference [-Wnull-dereference]
  855 |             for (auto* attachedPS = ps->attachedPS; attachedPS != nullptr; attachedPS = attachedPS->next)
      |                        ^~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unused-private-field’ may have been intended to silence earlier diagnostics
[121/219] Building CXX object CMakeFiles/openloco.dir/src/OpenLoco/S5/S5.cpp.o
In function ‘std::unique_ptr<OpenLoco::S5::SaveDetails> OpenLoco::S5::prepareSaveDetails(GameState&)’,
    inlined from ‘std::unique_ptr<OpenLoco::S5::S5File> OpenLoco::S5::prepareSaveFile(uint32_t, const std::vector<OpenLoco::ObjectHeader>&, const std::vector<OpenLoco::ObjectHeader>&)’ at /home/aaron/Repositories/OpenLoco/OpenLoco/src/OpenLoco/S5/S5.cpp:259:51,
    inlined from ‘bool OpenLoco::S5::save(OpenLoco::Stream&, uint32_t)’ at /home/aaron/Repositories/OpenLoco/OpenLoco/src/OpenLoco/S5/S5.cpp:315:78:
/home/aaron/Repositories/OpenLoco/OpenLoco/src/OpenLoco/S5/S5.cpp:163:21: warning: ‘char* strncpy(char*, const char*, size_t)’ specified bound 64 equals destination size [-Wstringop-truncation]
  163 |         std::strncpy(saveDetails->scenario, gameState.scenarioName, sizeof(saveDetails->scenario));
      |         ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unused-private-field’ may have been intended to silence earlier diagnostics
```

This is with g++ 12.1.0

No, the one warning is the following atm:
```
[171/294] Building CXX object src/OpenLoco/CMakeFiles/OpenLoco.dir/src/Paint/Paint.cpp.o
/home/aaron/Repositories/OpenLoco/OpenLoco/src/OpenLoco/src/Paint/Paint.cpp: In member function ‘OpenLoco::Ui::ViewportInteraction::InteractionArg OpenLoco::Paint::PaintSession::getNormalInteractionInfo(OpenLoco::Ui::ViewportInteraction::InteractionItemFlags)’:
/home/aaron/Repositories/OpenLoco/OpenLoco/src/OpenLoco/src/Paint/Paint.cpp:1097:24: warning: potential null pointer dereference [-Wnull-dereference]
 1097 |             for (auto* attachedPS = ps->attachedPS; attachedPS != nullptr; attachedPS = attachedPS->next)
      |                        ^~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unused-private-field’ may have been intended to silence earlier diagnostics
[294/294] Linking CXX executable OpenLoco
```

--- a/src/OpenLoco/src/Graphics/Gfx.h
+++ b/src/OpenLoco/src/Graphics/Gfx.h
@@ -1,6 +1,5 @@
 #pragma once
 
-#include "Graphics/PaletteMap.h"
 #include "ImageId.h"
 #include "Types.hpp"
 #include <OpenLoco/Core/EnumFlags.hpp>

--- a/src/OpenLoco/src/Paint/Paint.cpp
+++ b/src/OpenLoco/src/Paint/Paint.cpp
@@ -926,6 +926,7 @@ namespace OpenLoco::Paint
             // Draw any children this might have
             for (const auto* childPs = ps->children; childPs != nullptr; childPs = childPs->children)
             {
+                // assert(childPs->attachedPS == nullptr); Children can have attachments but we are skipping them to be investigated!
                 const bool shouldCullChild = shouldTryCullPaintStruct(*childPs, _viewFlags, getRotation(), _foregroundCullingHeight);
 
                 if (shouldCullChild)
@@ -1077,21 +1078,28 @@ namespace OpenLoco::Paint
 
         for (auto* ps = (*_paintHead)->basic.nextQuadrantPS; ps != nullptr; ps = ps->nextQuadrantPS)
         {
-            auto* tempPS = ps;
-            auto* nextPS = ps;
-            while (nextPS != nullptr)
+            // Check main paint struct
+            if (isSpriteInteractedWith(getRenderTarget(), ps->imageId, ps->vpPos))
             {
-                ps = nextPS;
-                if (isSpriteInteractedWith(getRenderTarget(), ps->imageId, ps->vpPos))
+                if (isPSSpriteTypeInFilter(ps->type, flags))
                 {
-                    if (isPSSpriteTypeInFilter(ps->type, flags))
-                    {
-                        info = { *ps };
-                    }
+                    info = { *ps };
                 }
-                nextPS = ps->children;
             }
 
+            // Check children paint structs
+            for (const auto* childPs = ps->children; childPs != nullptr; childPs = childPs->children)
+            {
+                if (isSpriteInteractedWith(getRenderTarget(), childPs->imageId, childPs->vpPos))
+                {
+                    if (isPSSpriteTypeInFilter(childPs->type, flags))
+                    {
+                        info = { *childPs };
+                    }
+                }
+            }
+
+            // Check attached to main paint struct
             for (auto* attachedPS = ps->attachedPS; attachedPS != nullptr; attachedPS = attachedPS->next)
             {
                 if (isSpriteInteractedWith(getRenderTarget(), attachedPS->imageId, attachedPS->vpPos + ps->vpPos))
@@ -1102,8 +1110,6 @@ namespace OpenLoco::Paint
                     }
                 }
             }
-
-            ps = tempPS;
         }
         return info;
     }

